client/suse/compat-suse.c

```diff
@@ -1766,12 +1766,20 @@ try_loopback(ni_suse_ifcfg_array_t *ifcfgs, ni_suse_ifcfg_t *ifcfg)
  * Handle infiniband devices
  */
 static ni_bool_t
-__maybe_infiniband(const char *ifname)
+maybe_infiniband(const char *ifname)
 {
+	char *end = NULL;
+	unsigned long tmp = 0;
+
 	if (ni_string_len(ifname) > 2 &&
 	    strncmp(ifname, "ib", 2) == 0 &&
-	    isdigit((unsigned char)ifname[2]))
-		return TRUE;
+	    isdigit((unsigned char)ifname[2]) &&
+	    (strtoul(ifname + 2, (char**) &end, 10) || TRUE)) {
+		if (*end == '\0')
+			return TRUE;
+		if (*end == '.' && ni_parse_ulong(end + 1, &tmp, 16) == 0)
+			return TRUE;
+	}
```

Comment:
… and please check device using `ni_netdev_name_is_valid(device)` before applying it.
Also, pkey and device are both mandatory for a child, thus we IMO need something like:
```c
        if (pkey || device) {
               unsigned long tmp = ~0UL;

                if (ni_parse_ulong(pkey, &tmp, 16) < 0 || tmp > 0xffff) {
                        ni_error("ifcfg-%s: Cannot find valid %s partition key number",
                                dev->name, ni_linktype_type_to_name(NI_IFTYPE_INFINIBAND_CHILD));
                        return -1;
                }
                if (!ni_netdev_name_is_valid(device)) {
                        ni_error("ifcfg-%s: Cannot find valid %s parent device name",
                                dev->name, ni_linktype_type_to_name(NI_IFTYPE_INFINIBAND_CHILD));
                        return -1;
                }

                // … set child iftype + apply pkey + device
        }
```

Reply:
We still need some check that there is a valid number after `ib`, which was made with `isdigit((unsigned char)ifname[2])`, otherwise it would match to `ib` and `ib.8001`, cause strtoul allow en empty strings to be 0.

Reply:
Ah, ok. I've missed this fact.